framework/src/task_manager/task_manager_set_new_broadcast_msg.c

```diff
@@ -0,0 +1,56 @@
+
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <task_manager/task_manager.h>
+#include "task_manager_internal.h"
+
+/****************************************************************************
+ * task_manager_set_new_broadcast_msg
+ ****************************************************************************/
+int task_manager_set_new_broadcast_msg(int msg, int timeout)
+{
+	int ret;
+	tm_request_t request_msg;
+
+	if (msg < TM_BROADCAST_PRE_DEFINE) {
+		return TM_INVALID_PARAM;
+	}
+
+	memset(&request_msg, 0, sizeof(tm_request_t));
+	request_msg.cmd = TASKMGRCMD_SET_NEW_BROADCAST_MSG;
+	request_msg.caller_pid = getpid();
+	request_msg.timeout = timeout;
+	request_msg.data = (void *)TM_ALLOC(sizeof(int));
```

Comment:
how about using handle instead of data? handle is not used for set new broadcast msg, so we can use.
 If we use handle, we don't have to alloc.
what's your opinion?

Reply:
In order to reduce an opportunity of misreading by other users, I think msg should be delivered at the data field.